### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by roo2**
Update some links and docs that @bangank36 noticed were broken during his onboarding.

### CODE REVIEW

File: docs/coding-guidelines/css.md [hunk 1]
```diff
@@ -157,7 +157,7 @@ Example of the above:
 
 - DO use `@extend` when in doubt, using the `%placeholder` syntax. This will produce leaner output.
 - DON'T use mixins for anything that doesn't accept an argument. This is `@extend` territory.
-- DO read [this article](http://miguelcamba.com/blog/2013/07/11/sass-placeholders-versus-mixins-and-extends/) if you don't understand `@extend`.
```
Comment (roo2) [inline docs/coding-guidelines/css.md:160]: link was dead

File: docs/coding-guidelines/html.md [hunk 2]
```diff
@@ -7,17 +7,17 @@ Writing great HTML is a craft.
 - Another developer should be able to read the html and have a clear picture of what everything is and their relationship, without scratching their head.
 - Another developer should be able to learn how a section works, and not have to relearn everything when they go to a different one. Consistency.
 - Another developer should be able to read a given class on an element and tell where in the structure it falls (sensible prefixes) and whether it's content represents a single entity or a list of entities.
-- The markup should not have style preferences, for the most part, so that changing how the CSS renders doesn't leave things with weird meanings (references to position or visuals, left, right, dark, blue).
+- The naming of markup and classes should be semantic and style-agnostic. For the most part, avoid references to position or visuals, (`-left`, `-right`, `-dark`, `-blue`, etc).
```
Comment (roo2) [inline docs/coding-guidelines/html.md:10]: just clarified this statement and added an example below

File: docs/coding-guidelines/typescript.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ norms and learn proper patterns and idioms; this team is there to help guide you
 
 ![](./typescript-dev-flow.dot.svg)
 
-Graph built with [`typescript-dev-flow.dot`](./typescript-dev-flow.dot)
+Graph built with [`typescript-dev-flow.dot`](https://github.com/Automattic/wp-calypso/blob/trunk/docs/coding-guidelines/typescript-dev-flow.dot)
```
Comment (roo2) [inline docs/coding-guidelines/typescript.md:23]: this link appears broken when viewed on https://wpcalypso.wordpress.com/devdocs/docs/coding-guidelines/typescript.md

File: docs/coding-guidelines/html.md [hunk 4]
```diff
@@ -37,24 +37,7 @@ As with JavaScript, HTML indentation should always reflect logical structure. Us
...
-				<h1 class="post-title">{ post.title }</h1>
-				<div class="post-content">{ post.content }</div>
-			</div>
-		);
-	}
-}
-```
-
-Incorrect:
+e.g.
```
Comment (roo2) [inline docs/coding-guidelines/html.md:40]: example was duplicated for both "correct" and "incorrect"

**Review by taipeicoder — APPROVED** [review]
LGTM 👍

**Pull request merged**